nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -1236,3 +1236,14 @@ def release(self) -> None:
     def run_callbacks(self) -> None:
         for callback in self.callbacks:
             callback()
+
+    def switch_root(self) -> None:
```

Comment:
Please add a doc-comment here.

Reply:
@tfc This is going to be a generically useful feature. We have a lot of systemd-initrd tests and *several* of them would be better off using this instead of some of the weird tricks they're using currently.

Reply:
For example, I recently did [this hacky thing](https://github.com/NixOS/nixpkgs/pull/251290/commits/1f845ebc343350c18e9d292c6999174b9c63fad2#diff-83c9f1bf18c273c33a829e698243902f47c958e5c08e6df69863a08bacb46fc2R33-R38) twice. It looks like this backdoor will be better.

Reply:
This is useful for more than a single test, indeed.

Reply:
Resolving, feel free to reopen if you disagree.